### PR TITLE
Internal Typing File for Type Aliases

### PR DESCRIPTION
**Pull request by GameDungeon**
## Overview: What does this pull request change?

Adds internal typing type aliases
Depreciates capture_mobject  

We have a lot of complex and heavily repeated type names. This pull request proposes to merge them into importable type aliases with simple, but helpful, names.

I also deprecated capture_mobject as it came to my attention during this pull and proved unnecessary.

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

**Comment by naveen521kk** [comment]
> I also deprecated capture_mobject as it came to my attention during this pull and proved unnecessary.

Create a new PR, don't mix stuff because it becomes hard while reviewing. Can you remove that change from this PR?

File: pyproject.toml [hunk 1]
```diff
@@ -76,6 +76,7 @@ flake8 = "^3.9.0"
 isort = "^5.8.0"
 pytest-xdist = "^2.2"
 psutil = ">=3.0"
+autopep8 = "^1.5.7"
```
Comment (naveen521kk) [inline pyproject.toml:79]: Why do you need this dependency?

Comment (GameDungeon) [reply pyproject.toml:79]: oops vscode added this for me.

File: manim/animation/animation.py [hunk 2]
```diff
@@ -368,7 +369,7 @@ def get_run_time(self) -> float:
 
     def set_rate_func(
         self,
-        rate_func: Callable[[float], float],
+        rate_func: "rate_function",
```
Comment (Darylgolden) [inline manim/animation/animation.py:372]: Should this be a string?

**Pull request closed without merging**